python/tk_multi_loader/delegate_publish_history.py

```diff
@@ -17,6 +17,107 @@
+        """
+        QtGui.QWidget.__init__(self, parent)
+
+        # make sure this widget isn't shown
+        self.setVisible(False)
+        
+        # set up the UI
+        self.ui = Ui_PublishHistoryWidget() 
+        self.ui.setupUi(self)
+        
+        # set up action menu
+        self._menu = QtGui.QMenu()   
+        self._actions = []             
+        self.ui.button.setMenu(self._menu)
+        self.ui.button.setVisible(False)
+        
+        # compute hilight colors
+        p = QtGui.QPalette()
+        highlight_col = p.color(QtGui.QPalette.Active, QtGui.QPalette.Highlight)
+        self._highlight_str = "rgb(%s, %s, %s)" % (highlight_col.red(), 
+                                                   highlight_col.green(), 
+                                                   highlight_col.blue())
+        self._transp_highlight_str = "rgba(%s, %s, %s, 25%%)" % (highlight_col.red(), 
+                                                                 highlight_col.green(), 
+                                                                 highlight_col.blue())
+        
+        
+    def set_actions(self, actions):
+        """
+        Adds a list of QActions to add to the actions menu for this widget.
+        
+        :param actions: List of QActions to add
+        """
+        if len(actions) == 0:
+            self.ui.button.setVisible(False)
+        else:
+            self.ui.button.setVisible(True)
+            self._actions = actions
+            for a in self._actions:
+                self._menu.addAction(a)
+                                    
+    def set_selected(self, selected):
+        """
+        Adjust the style sheet to indicate selection or not
+        
+        :param selected: True if selected, false if not
+        """
+        if selected:
+            self.ui.box.setStyleSheet("""#box {border-width: 2px; 
+                                                 border-color: %s; 
+                                                 border-style: solid; 
+                                                 background-color: %s}
+                                      """ % (self._highlight_str, self._transp_highlight_str))
+
+        else:
+            self.ui.box.setStyleSheet("")
+    
+    def set_thumbnail(self, pixmap):
+        """
+        Set a thumbnail given the current pixmap.
+        The pixmap must be 100x100 or it will appear squeezed
+        
+        :param pixmap: pixmap object to use
+        """
+        self.ui.thumbnail.setPixmap(pixmap)
+            
+    def set_text(self, header, body):
+        """
+        Populate the lines of text in the widget
+        
+        :param header: Header text as string
+        :param body: Body text as string
+        """
+        self.setToolTip("%s<br>%s" % (header, body))        
+        self.ui.header_label.setText(header)
+        self.ui.body_label.setText(body)
+
+    @staticmethod
+    def calculate_size():
```

Comment:
Feels like this should be possible by overloading the correct method on the widget e.g. size, sizeHint, minimumSize, minimumSizeHint...  not exactly sure which one though ;)